src/server/node_func.c

```diff
@@ -2907,3 +2912,38 @@ chk_vnode_pool (attribute *new, void *pobj, int actmode)
 	}
 	return PBSE_NONE;
 }
+
+/**
+ * @brief
+ * 		action routine for the queue's "partition" attribute
+ *
+ * @param[in]	pattr	-	attribute being set
+ * @param[in]	pobj	-	Object on which attribute is being set
+ * @param[in]	actmode	-	the mode of setting, recovery or just alter
+ *
+ * @return	error code
+ * @retval	PBSE_NONE	-	Success
+ * @retval	!PBSE_NONE	-	Failure
+ *
+ */
+int
+action_node_partition(attribute *pattr, void *pobj, int actmode)
+{
+	struct pbsnode *pnode;
+	pbs_queue 	*pq;
+
+	pnode = (pbsnode *)pobj;
+
+	if (pnode->nd_attr[(int)ND_ATR_Queue].at_flags & ATR_VFLAG_SET) {
+		pq = find_queuebyname(pnode->nd_attr[(int)ND_ATR_Queue].at_val.at_str);
+		if (pq == 0)
+			return PBSE_UNKQUE;
+		if (pq->qu_attr[QA_ATR_partition].at_flags & ATR_VFLAG_SET &&
+				pattr->at_flags & ATR_VFLAG_SET) {
+			if (strcmp(pq->qu_attr[QA_ATR_partition].at_val.at_str, pattr->at_val.at_str) != 0)
+				return PBSE_QUE_NOT_IN_PARTITION;
+		}
+	}
+
+	return PBSE_NONE;
+}
```

Comment:
As you said our current action function architecture does not handle the scenario of one attribute validation depending on another if both are set atomically or at once.  I think one of the following bugs (which says attribute validations are not happening when they are defined through vnodedef file)  that we have raised recently also suffers from this.  We also feel that this is the time for change of architecture for actions functions. 

https://pbspro.atlassian.net/browse/PP-925